src/test/java/address/guitests/FullSystemTest.java

```diff
@@ -0,0 +1,95 @@
+package address.guitests;
+
+import javafx.scene.input.KeyCode;
+import org.junit.Test;
+
+import static org.testfx.api.FxAssert.verifyThat;
+import static org.testfx.matcher.base.NodeMatchers.hasText;
+
+public class FullSystemTest extends GuiTestBase {
+    @Test
+    public void scenarioOne() {
+        // Attempt to create new tag, then cancel
+        clickOn("Tags").clickOn("New Tag");
+        verifyThat("#tagNameField", hasText(""));
+        push(KeyCode.ESCAPE);
+
+        // Attempt to create new tag
+        clickOn("Tags").clickOn("New Tag");
+        verifyThat("#tagNameField", hasText(""));
+
+        // Set name of tag to be "colleagues"
+        clickOn("#tagNameField").write("colleagues");
+        verifyThat("#tagNameField", hasText("colleagues"));
+        type(KeyCode.ENTER)
+
+        // Edit Hans Muster to John Tan, and edit details, including tags
+        .clickOn("Muster").type(KeyCode.E).sleep(1000)
+                .clickOn("#firstNameField").push(KeyCode.SHORTCUT, KeyCode.A).eraseText(1).write("John")
+                .clickOn("#lastNameField").eraseText(6).write("Tan")
+                .clickOn("#cityField").write("Singapore")
+                .clickOn("#githubUserNameField").write("john123");
+        verifyThat("#firstNameField", hasText("John"));
+        verifyThat("#lastNameField", hasText("Tan"));
+        verifyThat("#cityField", hasText("Singapore"));
+        verifyThat("#githubUserNameField", hasText("john123"));
+                clickOn("#tagList")
```

Comment:
Indented to show that we are still performing the operation in the `PersonEditDialog` (like editing the other fields). Not too sure what's a good style for this since we have to cut off the chain at some points to verify behaviour (if not it is really difficult to debug).